pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -417,25 +417,39 @@ func (m *kubeGenericRuntimeManager) podSandboxChanged(pod *v1.Pod, podStatus *ku
 		}
 	}
 
-	// Needs to create a new sandbox when readySandboxCount > 1 or the ready sandbox is not the latest one.
+	//check the pod is succeeded
+	succeededContainerCount := 0
+	for _, container := range pod.Spec.Containers {
+		for _, status := range podStatus.ContainerStatuses {
+			if container.Name == status.Name {
+				if status.State == kubecontainer.ContainerStateExited && status.ExitCode == 0 {
+					succeededContainerCount++
+				}
+			}
+		}
+	}
+
+	// Needs to create a new sandbox when readySandboxCount > 1.
 	sandboxStatus := podStatus.SandboxStatuses[0]
 	if readySandboxCount > 1 {
 		klog.V(2).Infof("More than 1 sandboxes for pod %q are ready. Need to reconcile them", format.Pod(pod))
 		return true, sandboxStatus.Metadata.Attempt + 1, sandboxStatus.Id
 	}
-	if sandboxStatus.State != runtimeapi.PodSandboxState_SANDBOX_READY {
+
+	//Needs to create a new when not all container succeeded and the ready sandbox is not the latest one
```

Comment:
Space after //
create a new what?
```suggestion
	// Needs to create a new ??? when not all container succeeded and the ready sandbox is not the latest one
```